### PR TITLE
Upgraded React Native, Onfido SDK's and simplified integration

### PR DESCRIPTION
**Pull request by anuragajwani**
# Problem
The React Native integration does not reflect integration with the latest versions of Onfido's Mobile SDK's and React Native.

Furthermore the integration for both platforms was different.

# Solution
Upgraded react native versions, Onfido's Mobile SDK's and simplified React Native integration with Onfido's Mobile SDK's

# Checklist
put n/a if item is not relevant to PR changes
[x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
[n/a] Have new automated tests been implemented?
[n/a] Have new manual tests been written down?
[n/a] Have tests passed locally?
[n/a] Have any new strings been translated?

### CODE REVIEW

File: android/app/src/main/java/com/reactnativesampleapp/OnfidoSDK.java [hunk 1]
```diff
@@ -42,6 +43,11 @@ public void userCompleted(Applicant applicant, Captures captures) {
                 public void userExited(ExitCode exitCode, Applicant applicant) {
                     mErrorCallback.invoke(exitCode.toString());
                 }
+
+                @Override
+                public void onError(OnfidoException e, Applicant applicant) {
+                    mErrorCallback.invoke("Onfido SDK errored");
```
Comment (sergiomarqmoura) [inline android/app/src/main/java/com/reactnativesampleapp/OnfidoSDK.java]: Can we replace `"Onfido SDK errored"` by `e.getMessage()`? It gives the integrator more context on what happened

File: android/app/src/main/java/com/reactnativesampleapp/OnfidoSDK.java [hunk 2]
```diff
@@ -58,7 +64,10 @@ public String getName() {
     }
 
     @ReactMethod
-    public void startSDK(Callback successCallback, Callback errorCallback) {
+    public void startSDK(String token, String applicantId, Callback successCallback, Callback errorCallback) {
+
+        System.out.printf(String.format("token=%s, applicantId=%s", token, applicantId));
```
Comment (sergiomarqmoura) [inline android/app/src/main/java/com/reactnativesampleapp/OnfidoSDK.java]: Can we remove this? Logging tokens is not a good idea

**Review by sergiomarqmoura — CHANGES_REQUESTED** [review]
Nice work!

File: android/app/src/main/java/com/reactnativesampleapp/OnfidoSDK.java (hunk 2 above)
Comment (anuragajwani) [reply android/app/src/main/java/com/reactnativesampleapp/OnfidoSDK.java]: this added for debugging but forgot to remove when cleaning up

File: package.json [hunk 3]
```diff
@@ -7,15 +7,17 @@
 		"test": "jest"
 	},
 	"dependencies": {
-		"react": "16.0.0",
-		"react-native": "0.50.3",
+		"react": "16.6.0-alpha.8af6728",
```
Comment (rfreitas) [inline package.json:10]: interesting that upgrades to alpha

**Review by rfreitas — APPROVED** [review]
Great job! I share sergio's comment on the logs, but nothing extra for me, so approving.

**Review by sergiomarqmoura — APPROVED** [review]

**Pull request merged**